### PR TITLE
Try using hoistLogicT

### PR DESCRIPTION
**Pull request by treeowl**
Do we get the same (or equivalent) results when we `hoistLogicT` instead of running the computation and then scattering? It will depend on what the actions are up to.

Fixes #nnnn

### Estimate:

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [ ] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [ ] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [ ] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [ ] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

**Comment by treeowl** [comment]
Does anyone know if there's a reason the code was written to collect all the results and then scatter them, rather than working in a streaming fashion? Changing that doesn't break any of the tests, but I'd like to be sure I'm not doing something really unexpected here.

**Comment by treeowl** [comment]
The KEVM test saw no perf difference. I think this is cleaner from a code standpoint; I can't speak to what makes more sense algorithmically when it comes to the effect ordering.

**Review by ana-pantilie — APPROVED** [review]

File: kore/src/Logic.hs [hunk 1]
```diff
@@ -34,8 +36,18 @@ mapLogicT ::
     (forall x. m x -> n x) ->
     LogicT m a ->
     LogicT n a
-mapLogicT nat acts = (lift . nat) (observeAllT acts) >>= scatter
+mapLogicT nat acts = hoistLogicT nat acts
 {-# INLINE mapLogicT #-}
+#if !MIN_VERSION_logict(0,8,0)
```
Comment (ana-pantilie) [inline kore/src/Logic.hs:41]: From @jberthold : we should remove this once we are definitely on the `0.8` version

**Pull request merged**